app/controllers/departments_controller.ts

```diff
@@ -0,0 +1,40 @@
+import type { HttpContext } from "@adonisjs/core/http";
+
+import Department from "#models/department";
+import { showValidator } from "#validators/show";
+
+export default class DepartmentsController {
+  protected readonly relations = ["fieldOfStudy", "departmentLink"];
+
+  async index({ request }: HttpContext) {
+    const departments = await Department.query().withScopes((scopes) => {
+      scopes.handleSearchQuery(
+        request.only([
+          "id",
+          "name",
+          "code",
+          "betterCode",
+          "createdAt",
+          "updatedAt",
+        ]),
+      );
+      scopes.includeRelations(request.only(this.relations));
+      scopes.handleSortQuery(request.input("sort"));
+    });
+    return { data: departments };
+  }
```

Comment:
I think searching by address might be useful, which this code currently does not allow.
As I said in my previous review, it's quicker to only list fields that should not be searchable, as you'll likely have more searchable fields than those that are not.
In the case of this model, I think only the gradient colors and logo key should not be searchable.

Also this endpoint is missing the option for result pagination.
And the relation scope has a different name than in all other models.